lapce-ui/src/settings.rs

```diff
@@ -138,27 +139,28 @@ impl LapceSettingsPanel {
     }
 
     fn update_plugins(&mut self, ctx: &mut EventCtx, data: &LapceTabData) {
-        let current_keys: Vec<LapceSettingsKind> =
-            self.children.keys().cloned().collect();
-        for kind in current_keys {
+        let drained = self.children.drain_filter(|kind, _| {
             if let LapceSettingsKind::Plugin(volt_id) = &kind {
-                if !data.plugin.installed.keys().contains(&volt_id) {
-                    self.children.remove(&kind);
-                    ctx.children_changed();
-                    if self.active == kind {
-                        self.active = LapceSettingsKind::Core;
-                        self.switcher
-                            .widget_mut()
-                            .child_mut()
-                            .set_active(self.active.clone(), data);
-                    }
-                }
+                !data.plugin.installed.contains_key(volt_id)
+            } else {
+                false
+            }
+        });
+        for (kind, widget_pod) in drained {
+            drop(widget_pod);
```

Comment:
Is this drop necessary? Wouldn't it get dropped as drained is dropped anyways?

Reply:
The actual calling sequence of element added/removed and ```ctx.children_changed()``` doesn't really matter. 

```ctx.children_changed()``` only sets a flag and used for the next paint cycle.

Reply:
Yes I understand. I'm generally going to throw out the multiple call to this function in a loop and just have some extra flag or something like that. I'll think about it after work.

Reply:
Switched to using `retain` instead of `drain_filter` and `drop`